WordPress/Classes/Services/PostAutoUploadInteractor.swift

```diff
@@ -64,4 +64,14 @@ final class PostAutoUploadInteractor {
 
         return !PostAutoUploadInteractor.allowedStatuses.contains(status)
     }
+
+    /// Returns true if any of the following conditions are true:
+    ///
+    /// * The post is a draft.
+    /// * The post failed to upload and has local changes but the user canceled auto-uploading
+    /// * The upload failed and the user cannot Cancel it anymore. This happens when we reached the maximum number of retries.
+    func canPublish(of post: AbstractPost) -> Bool {
```

Comment:
This seems to be working well. 👍 

So, when I was working on `PostAutoUploadInteractor`, I was aiming for it to only contain logic for the auto-uploading flow: 

- What actions are performed
- If the action can be canceled

You can see that `canRetryUpload(of:)` doesn't fit this. Because, as described in the comment, it is supposed to be temporary. 😐 I figured this was okay for now since it's supposed to be removed and it was easier to find. But maybe that was bad judgement on my part. I could have used `#warning()` instead.

I believe `canPublish(of:)` is also the same. It's not really part of auto-upload. It uses the `PostAutoUploadInteractor` logic to derive its complete business logic. With this, we should just keep it inside `PostCardStatusViewModel`:

```swift
var canPublish: Bool {
    let isNotCancelablePublishedWithFailedToUploadChanges: Bool = post.isFailed && post.hasLocalChanges() && !canCancelAutoUpload(of: post)
    return post.isDraft() || isNotCancelablePublishedWithFailedToUploadChanges
}
```

Reply:
I''l move it to the view model.